src/test/fakes.rs

```diff
@@ -76,15 +77,37 @@ impl<'db> FakeRelease<'db> {
         self
     }
 
+    pub(crate) fn file<M, P, D>(mut self, mimetype: M, path: P, data: D) -> Self
+        where M: Into<String>,
+              P: Into<String>,
+              D: Into<Vec<u8>>,
+        {
+        let (mimetype, path, data) = (mimetype.into(), path.into(), data.into());
+        if mimetype.contains('"') || path.contains('"') {
+            panic!("invalid mimetype or path: {}, {}", mimetype, path);
+        }
+        self.files.push((mimetype, path, data));
+
+        self
+    }
+
     pub(crate) fn create(self) -> Result<i32, Error> {
         let tempdir = tempdir::TempDir::new("docs.rs-fake")?;
 
+        let files: Vec<_> = self.files.into_iter().map(|(mimetype, path, data)| {
+            let file = tempdir.path().join(&path);
+            std::fs::write(file, data)?;
+
+            let metadata = format!("[\"{}\", \"{}\"]", mimetype, path);
+            Ok(Json::from_str(&metadata).expect("should be valid json"))
+        }).collect::<Result<_, Error>>()?;
+
         let release_id = crate::db::add_package_into_database(
             &self.db.conn(),
             &self.package,
             tempdir.path(),
             &self.build_result,
-            self.files,
+            if files.is_empty() { None } else { Some(Json::Array(files)) },
```

Comment:
Hmm, I'm pretty sure the `files` here refers to source files, not to the rustdoc documentation.

Reply:
Oh whoops! I need to reshuffle some things then.

Reply:
Also, we should probably document that. It's referred to as 'files' all through the code until it goes into the DB and it's hard to find again.

Reply:
Definitely.

Reply:
I did some refactoring, this should be fixed now.